core/lib/dal/src/blocks_dal.rs

```diff
@@ -2334,6 +2355,84 @@ impl BlocksDal<'_, '_> {
 
         Ok(results.into_iter().map(L::from).collect())
     }
+
+    pub async fn get_last_l2_block_bloom(&mut self) -> DalResult<Option<Bloom>> {
+        let row = sqlx::query!(
+            r#"
+            SELECT
+                logs_bloom
+            FROM
+                miniblocks
+            ORDER BY
+                number DESC
+            LIMIT
+                1
+            "#,
+        )
+        .instrument("get_last_l2_block_bloom")
+        .fetch_optional(self.storage)
+        .await?;
+
+        Ok(row
+            .and_then(|row| row.logs_bloom)
+            .map(|b| Bloom::from_slice(&b)))
+    }
+
+    pub async fn get_max_l2_block_without_bloom(&mut self) -> DalResult<Option<L2BlockNumber>> {
+        let row = sqlx::query!(
+            r#"
+            SELECT
+                MAX(number) AS "max?"
+            FROM
+                miniblocks
+            WHERE
+                logs_bloom IS NULL
+            "#,
+        )
+        .instrument("get_max_l2_block_without_bloom")
+        .fetch_one(self.storage)
+        .await?;
+
+        Ok(row.max.map(|n| L2BlockNumber(n as u32)))
+    }
+
+    pub async fn range_update_logs_bloom(
+        &mut self,
+        from_l2_block: L2BlockNumber,
+        to_l2_block: L2BlockNumber,
```

Comment:
Bikeshedding: IMO, it would be more idiomatic to use an inclusive range; ditto for other new queries. In this particular case, it's possible to skip `to_l2_block` since it can be trivially restored from other supplied data.